tests/distributed/test_deterministic_all_reduce.py

```
@@ -0,0 +1,116 @@
+                    leaves_per_rank = _MAX_WORLD_SIZE // tp_size
+                    start = group_rank * leaves_per_rank
+                    for dtype in (torch.float32, torch.float16, torch.bfloat16):
+                        generator = torch.Generator().manual_seed(20260815)
+                        leaves_tensor = torch.randn(
+                            _MAX_WORLD_SIZE,
+                            257,
+                            dtype=torch.float32,
+                            generator=generator,
+                        ).to(device=device, dtype=dtype)
+                        leaves = list(leaves_tensor.unbind())
+                        input = _fixed_tree_reference(leaves[start : start + leaves_per_rank])
```

Ruff (0.16.1)

[error] Variable `input` is shadowing a Python builtin

(A001)